core/src/main/scala/org/http4s/client/jdkhttpclient/JdkHttpClient.scala

```diff
@@ -50,15 +49,15 @@ object JdkHttpClient {
     */
   def apply[F[_]](
       jdkHttpClient: HttpClient,
-      ignoredHeaders: Set[CaseInsensitiveString] = restrictedHeaders
-  )(implicit F: ConcurrentEffect[F], CS: ContextShift[F]): Client[F] = {
+      ignoredHeaders: Set[CIString] = restrictedHeaders
+  )(implicit F: Async[F]): Resource[F, Client[F]] = Dispatcher[F].map { dispatcher =>
```

Comment:
I think one of our other backends (Jetty? I can't remember) called this `fromHttpClient`, since it no longer returns the type, but rather a `Resource`.  The thinking being that `apply` on a companion object should return an instance of the companion class.  But `JdkHttpClient` isn't a `Client`, so maybe that guideline isn't really relevant here.

They also call the thing that returns a resource `.resource`, but those also follow the builder pattern that this one isn't complicated enough to follow.

I'm bikeshedding.  This is probably all fine as is.